LibSWBF2.NET/Types/Vector2.cs

```diff
@@ -3,20 +3,26 @@
 
 namespace LibSWBF2.Types
 {
+    [StructLayout(LayoutKind.Sequential, Pack=4)]
     public struct Vector2
     {
         public float X;
         public float Y;
 
-        public Vector2(IntPtr nativePtr)
+        public unsafe Vector2(IntPtr nativePtr) : this()
         {
-        	APIWrapper.Vector2_FromPtr(nativePtr, out X, out Y);
+            if (nativePtr != IntPtr.Zero && nativePtr != null)
+            {
+                float* fPtr = (float*) nativePtr.ToPointer();
+                X = *fPtr;
+                Y = *(fPtr + 1);                
+            }
         }
 
```

Comment:
Haven't tested this on Windows yet.  Seems like a bad idea, but works w/Mono.

Reply:
Will test it shortly